code/modules/mob/living/carbon/human/diona_gestalt.dm

```diff
@@ -359,6 +359,47 @@
 	visible_message(SPAN_WARNING("\The [src] quivers slightly, then splits apart with a wet slithering noise."))
 	qdel(src)
 
+/**
+ * This verb lets the Diona anchor themselves to the ground, giving them magboot properties.
+ */
+/mob/living/carbon/human/proc/root_to_ground()
+	set name = "Extend Roots"
+	set desc = "Extend your roots to keep yourself upright in case of pressure changes."
+
+	if(use_check_and_message(src))
+		return
+
+	if(nutrition <= 50)
+		to_chat(src, SPAN_WARNING("You lack nutrition to perform this action!"))
+		return
+
+	if(DS.stored_energy <= 20)
+		to_chat(src, SPAN_WARNING("You lack energy to perform this action!"))
+		return
+
+	if(last_special > world.time)
+		to_chat(src, SPAN_WARNING("You've used an ability too recently! Wait a bit."))
+		return
+
+	if(!has_organ(BP_R_FOOT) && !has_organ(BP_L_FOOT))
+		to_chat(src, SPAN_WARNING("You have no supporting limbs to extend your roots from!"))
+		return
+
+	if(isspaceturf(get_turf(src)))
+		to_chat(src, SPAN_WARNING("You can't do that here!"))
+		return
```

Comment:
I currently only have my phone available, so searching through the code is rather complicated.

It seems that all these checks are only performed once.

However if these checks are required to enable the function, they should also be performed regularly while using it.

There should also be some cost to using the "magboot" function, either in the form of decreased movement speed or increased hunger (or something else?)
Is that already handled somewhere else (by the trait?)?